ERCS/erc-7760.md

```diff
@@ -0,0 +1,511 @@
+---
+eip: 7760
+title: Minimal Upgradeable Proxies
+description: Minimal upgradeable proxies with immutable arguments and support for onchain implementation queries
+author: Atarpara (@Atarpara), JT Riley (@jtriley-eth), xiaobaiskill (@xiaobaiskill), Vectorized (@Vectorized)
+discussions-to: https://ethereum-magicians.org/t/erc-7760-minimal-upgradeable-proxies/20868
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-19
+requires: 1967
+---
+
+## Abstract
+
+This standard defines minimal [ERC-1967](./eip-1967.md) proxies for three patterns: (1) transparent, (2) UUPS, (3) beacon. The proxies support optional immutable arguments which are appended to the end of their runtime bytecode. Additional variants which support onchain implementation querying are provided.
+
+## Motivation
+
+Having standardized minimal bytecode for upgradeable proxies enables the following:
```

Comment:
Can you provide details about how these bytecodes are generated so people can verify the correctness and minimal-ness?

Reply:
If we are to be pendantic, ERC-1167 is NOT minimal at the time of finalization. The 0age proxy is more minimal in both runtime gas and bytecode size.

The transparent pattern is not strictly minimal for the upgrade path. But since it has been used in the wild, including the version that has been used is more beneficial than including the theoretically more minimal version. The purpose of this standard is to provide a document for block explorers to auto verify. Support for existing proxies outweighs the gas saved.

Also, users who want to squeeze out every single bit of runtime gas savings and bytecode size will use the UUPS variant.

With all this in consideration, the transparent variant is minimal-enough.